### PR TITLE
Fix memory leak in server

### PR DESCRIPTION
**Pull request by AlexTugarev**
This PR fixes two things:
1. it resolves an issue with dynamic auth providers being initialized in a 2s loop, if `host` attribute stored in DB is not lower case. DB test included.
2. it also sanitizes log statements including big/unwanted/boring objects as payload to be serialized, thus pulling down performance of server.

### How to test
1. Create a Git Integration with garbage details, the only thing of interest is that the host should contain upper case char. 
2. Check debug logs to not include sequences of {initializing provider1, disposing provider1}

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
@AlexTugarev: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by AlexTugarev** [comment]
/werft run

timeout in tests

File: components/server/src/auth/host-context-provider-impl.ts [hunk 1]
```diff
@@ -112,10 +112,10 @@ export class HostContextProviderImpl implements HostContextProvider {
         // remove obsolete entries
         const currentHosts = new Set(await this.authProviderService.getAllAuthProviderHosts());
         ctx.span?.setTag("updateDynamicHosts.currentHostProviders", currentHosts.size);
+        // HINT: values of `currentHosts` are expected to be lower case
         const tobeRemoved = [...this.dynamicHosts.keys()].filter((h) => !currentHosts.has(h));
         for (const host of tobeRemoved) {
-            const hostContext = this.dynamicHosts.get(host);
-            log.debug("Disposing dynamic Auth Provider: " + host, { host, hostContext });
```
Comment (AlexTugarev) [inline components/server/src/auth/host-context-provider-impl.ts:118]: BTW this was the kick-off of for the log sanitizer commit.

**Comment by geropl** [comment]
Tested as advertised, and only saw these two entries with `host` being normalized to lower case:
image

**Review by geropl — APPROVED** [review]
Code LGTM, tested and seems to work.

Thx for the test! :pray:

**Pull request merged**